### DISCUSSION TITLE
tap operator's callback function should have an index parameter (like the map operator)

### DISCUSSION THREAD

**Discussion by iuliust** [ReactiveX/rxjs#6702] [Ideas / Feature request]
When using the map operator, one can get the index of current event like so:
```typescript
myObservable.pipe(
  map((value, index) => {
    console.log(`This is the ${ index }th call since subscription.`);
  }),
).subscribe();
```

This is quite useful.
Unfortunately, the tap operator does not offer this feature. I think it would be a useful addition to the framework.

**Comment by kwonoj** [comment]
Dupe of https://github.com/ReactiveX/rxjs/issues/5461#issuecomment-635638345, Core's tap fn signature is desinged to be compatible with subscriber explicitly: https://github.com/ReactiveX/rxjs/blob/master/src/internal/operators/tap.ts#L115-L117

Creating userland operator to achieve those beahvior is easy enough as well.
